web/modules/custom/collection/collection.module

```diff
@@ -15,3 +16,20 @@ function collection_form_block_form_alter(&$form, FormStateInterface $form_state
   // good use cases to show a block on all pages, except on collection pages.
   $form['visibility']['collection']['negate']['#access'] = FALSE;
 }
+
+/**
+ * Implements hook_entity_base_field_info_alter().
+ */
+function collection_entity_base_field_info_alter(&$fields, EntityTypeInterface $entity_type) {
+  /** @var \Drupal\Core\Field\FieldDefinitionInterface[] $fields */
+  if ($entity_type->id() == 'rdf_entity') {
+    if (!empty($fields['label'])) {
+      $bundles = ['collection'];
+      if ($constraint = $fields['label']->getConstraint('UniqueFieldInBundle')) {
+        $bundles = array_merge($bundles, $constraint['bundles']);
+      }
+
+      $fields['label']->addConstraint('UniqueFieldInBundle', ['bundles' => $bundles]);
+    }
+  }
```

Comment:
This seems strange, altering the base fields will make the constraint available on ALL the bundles. It would be better to do this on a per-bundle basis. This is certainly possible since you can even assign validation constraints through the UI (e.g. make a field required).